openstates/in/bills.py

```diff
@@ -45,198 +45,185 @@ class INBillScraper(BillScraper):
-            bill_id = a.text_content().strip()
-            bill_chamber = ('upper' if bill_id[0] in 'SJ' else 'lower')
-            bill_url = a.attrib['href']
-            bill_title = a.xpath('string(./following-sibling::em)').strip()
-            if bill_title == "":
-                self.logger.warning("Bill %s has no title" % bill_id)
-                continue
-            self.scrape_bill(bill_chamber, term, bill_id, bill_url, bill_title)
-
-        url = 'http://iga.in.gov/legislative/%s/bills/' % term
-        self.logger.info('GET ' + url)
-        html = self.get(url).text
-        doc = lxml.html.fromstring(html)
-        doc.make_links_absolute(url)
-
-
-        # First scrape subjects and any bills found on those pages.
-        for subject_url, subject in self.generate_subjects(term):
-            subject_bills = self.generate_subject_bills(subject_url)
-            for bill_id, bill_url, bill_title in subject_bills:
-                chamber = 'upper' if bill_id[0] == 'S' else 'lower'
-                self.scrape_bill(
-                    chamber, term, bill_id, bill_url, bill_title, subject=subject)
-                seen_bill_ids.add(bill_id)
-
-        # Then hit bill index page to catch any uncategorized bills.
-        uls = doc.xpath('//ul[contains(@class, "clean-list")]')
-
-        for chamber, ul in zip(chambers, uls):
-            for li in ul.xpath('li'):
-                bill_id = li.xpath('string(a/strong)')
-                if bill_id in seen_bill_ids:
-                    continue
-                bill_url = li.xpath('string(a/@href)')
-                try:
-                    bill_title = li.xpath('a/strong')[0].tail.rstrip().lstrip(': ')
-                except IndexError:
-                    continue
+    def make_html_source(self,session,bill_id):
+        url = "http://iga.in.gov/legislative/{}/".format(session)
+        urls = {
+            "hb":"bills/house/",
+            "hr":"resolutions/house/simple/",
+            "hcr":"resolutions/house/concurrent",
+            "hjr":"resolutions/house/joint",
+            "hc":"resolutions/house/concurrent",
+            "hj":"resolutions/house/joint",
+            "sb":"bills/senate/",
+            "sr":"resolutions/senate/simple/",
+            "scr":"resolutions/senate/concurrent",
+            "sjr":"resolutions/senate/joint",
+            "sc":"resolutions/senate/concurrent",
+            "sj":"resolutions/senate/joint"
+            }
+        bill_id = bill_id.lower()
+        try:
+            int(bill_id[2])
+        except ValueError:
+            bill_prefix = bill_id[:3]
+            bill_num = bill_id[3:]
+        else:
+            bill_prefix = bill_id[:2]
+            bill_num = bill_id[2:]
 
-                if bill_title.strip() == "":
+        try:
+            url += urls[bill_prefix]
+        except KeyError:
+            raise AssertionError("Unknown bill type {}, don't know how to make url".format(bill_id))
+        url += bill_num
```

Comment:
For URLs, `bill_num` needs to have no leading zeros. Eg, `http://iga.in.gov/legislative/2015/bills/senate/15` instead of `http://iga.in.gov/legislative/2015/bills/senate/0015`.